tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py

```diff
@@ -108,107 +63,144 @@ def parameters(cls):
             cur_num = len(labels)
             assert len(scores) == cur_num
 
-            coco_cats = [map_pred_label_id_to_coco_cat_id[lbl] for lbl in labels]
-            for (s, cur_cat) in zip(scores, coco_cats):
+            for (s, cur_cat) in zip(scores, labels):
                 prediction_data_to_store.append({
                     'image_id': cur_img_id,
                     'score': s,
                     'category_id': cur_cat,
                     '_image_name_from_dataset': cur_name,
                 })
-            iou_specific_converter = iou_specific_processing.get(iou_type)
-            if iou_specific_converter is None:
-                raise ValueError("unknown iou type: '{}'".format(iou_type))
-            prediction_data_to_store = iou_specific_converter(prediction_data_to_store, pred)
+            prediction_data_to_store = self._iou_type_data_to_coo(prediction_data_to_store, pred)
             coco_data_to_store.extend(prediction_data_to_store)
 
         return coco_data_to_store
 
+    def _iou_type_specific_coco_annotation(self, annotation_data_to_store, annotation):
+        annotation_data_to_store = self._iou_type_data_to_coo(annotation_data_to_store, annotation)
+        return annotation_data_to_store
+
+    def _prepare_data_for_annotation_file(
+            self, annotations, dataset_label_map
+    ):
+        coco_annotation_to_store = []
+        coco_category_to_store = []
+        coco_image_to_store = []
+        label_map = set()
+        count = 0
+        for annotation in annotations:
+            annotation_data_to_store = []
+            cur_name = pathlib.PurePath(annotation.identifier).name
+            cur_img_id = int(cur_name.split(".")[0])
+
+            if self.iou_type != 'segm':
+                labels = annotation.labels.tolist()
+            else:
+                labels = annotation.labels
+
+            if annotation.metadata.get('iscrowd'):
+                iscrowds = annotation.metadata.get('iscrowd')
+            else:
+                iscrowds = [0]*annotation.size
+
+            for cur_cat, iscrowd in zip(labels, iscrowds):
+                annotation_data_to_store.append({
+                    'id': count,
+                    'image_id': cur_img_id,
+                    'category_id': cur_cat,
+                    '_image_name_from_dataset': cur_name,
+                    'iscrowd': iscrowd
+                })
+                count += 1
+                label_map.add(cur_cat)
+
+            annotation_data_to_store = self._iou_type_specific_coco_annotation(annotation_data_to_store, annotation)
+            coco_annotation_to_store.extend(annotation_data_to_store)
+
+            coco_image_to_store.append({
+                'id': cur_img_id,
+                'file_name': cur_name,
+                'width': annotation.metadata.get('image_size')[0][0],
+                'height': annotation.metadata.get('image_size')[0][1]
+            })
+        for cat in label_map:
```

Comment:
why do you need these changes?

dataset_label_map should contains full dataset label map in format id: cat_name. You need only skip background if it present in dataset (there is specific field in the dataset meta `background_label`, which should be checked), so no need in category storing in the cycle

Reply:
please review again. label_map set has been removed. dataset label map is used to create categories list